posthog/api/plugin.py

```diff
@@ -89,16 +119,27 @@ def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Plugin:
         return plugin
 
     def update(self, plugin: Plugin, validated_data: Dict, *args: Any, **kwargs: Any) -> Plugin:  # type: ignore
-        if not can_install_plugins_via_api(self.context["organization_id"]):
-            raise ValidationError("Plugin upgrades via the web are disabled!")
-        if plugin.plugin_type != Plugin.PluginType.SOURCE:
-            validated_data = self._update_validated_data_from_url({}, plugin.url)
+        context_organization = self.context.get("organization") or Organization.objects.get(
+            id=self.context["organization_id"]
+        )
+        if plugin.organization != context_organization:
```

Comment:
Q: is this not duplicating PluginOwnershipPermission and PluginsAccessLevelPermission? Why?